src/hm-admin/services/core/CoreDefinition.service.ts

```diff
@@ -0,0 +1,95 @@
+
+import {Http} from '@angular/http';
+import {Injectable} from '@angular/core';
+import 'rxjs/add/operator/map';
+import {Observable} from 'rxjs';
+
+@Injectable()
+export class CoreDefinition {
+
+    public coreDefinition: any;
+
+    constructor(private http: Http) {}
+
+    /**
+     * Get the CoreDefinition, if it exists from Localstorage, if not from HTTP
+     *
+     * @param {string} url
+     *
+     * @returns {Observable}
+     */
+    getCoreDefinition(url) {
+        let definition;
+        this.coreDefinition = this.getCacheCoreDefinition();
+
+        if (this.coreDefinition === null) {
+            definition = this.fetchCoreDefinition(url);
+            this.setCacheCoreDefinition(definition);
+        }
+        return this.coreDefinition;
+    }
+
+    /**
+     * get the Definition from url
+     *
+     * @param {string} url
+     *
+     * @returns {Observable}
+     */
+    fetchCoreDefinition(url): Observable<any> {
+        let coreObject: Definition = {};
+        return this.http.get(url)
+         .map(res => res.json())
+         .map(res => res['@context'])
+         .map(res => {
+             for (let key in res) {
+                let value = res[key];
+                if (value instanceof Object && typeof value['@id'] !== 'undefined') {
+                    value = value['@id'];
+                }
+                 this.add(key, value, coreObject);
```

Comment:
why passing and object when you already know that the target will always be `this.definitions`?